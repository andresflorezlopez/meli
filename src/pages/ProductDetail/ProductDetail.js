import React from "react";
import Button from "@components/Button/Button";
import MainLayout from "@layouts/MainLayout/MainLayout";
import Typography from "@components/Typography/Typography";
import {
  StyledHead,
  StyledImage,
  StyledInformation,
  StyledDescription,
  StyledImageContainer
} from "./ProductDetail.styled";
const ProductDetail = () => {
  const product = {
    title: "Deco master test Oxford magic",
    price: "1.900",
    condition: "Nuevo",
    sold_quantity: "200",
    image: "https://picsum.photos/id/180/680/680",
    description:
      "Elit aliquip aliquip in ullamco consequat voluptate. Ea ipsum excepteur nostrud proident. Fugiat irure aliqua ad culpa quis Lorem eiusmod in veniam Lorem. Elit aliquip aliquip in ullamco consequat voluptate. Ea ipsum excepteur nostrud proident. Fugiat irure aliqua ad culpa quis Lorem eiusmod in veniam Lorem."
  };
  return (
    <MainLayout>
      <StyledHead>
        <StyledImageContainer>
          <StyledImage src={product.image} />
        </StyledImageContainer>
        <StyledInformation>
          <Typography type="caption">
            {`${product.condition} - ${product.sold_quantity} vendidos`}
          </Typography>
          <Typography type="h3" fontWeight="800">
            {product.title}
          </Typography>
          <Typography type="h1">$ {product.price}</Typography>
          <Button onClick={() => console.log("Coming soon")}>Comprar</Button>
        </StyledInformation>
      </StyledHead>
      <StyledDescription>
        <Typography type="h2">Descripcion del product</Typography>
        <Typography>{product.description}</Typography>
      </StyledDescription>
    </MainLayout>
  );
};

export default ProductDetail;
