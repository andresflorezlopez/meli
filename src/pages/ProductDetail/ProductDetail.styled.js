import styled from "styled-components";
import shippingIcon from "@assets/images/shipping.png";

export const StyledHead = styled.div`
  display: flex;
  flex-direction: row;
  padding: ${({ theme }) => theme.spacing(4)};
  ${({ theme }) => theme.breakpoints.sm} {
    flex-direction: column;
  }
`;

export const StyledImageContainer = styled.div`
  flex: 6;
  padding-right: ${({ theme }) => theme.spacing(2)};
`;

export const StyledImage = styled.img`
  border-radius: ${({ theme }) => theme.borderRadius.normal};
  max-width: 680px;
  height: auto;
  width: 100%;
`;

export const StyledInformation = styled.div`
  flex: 3;
  h3 {
    padding-top: ${({ theme }) => theme.spacing(2)};
    max-width: 60%;
  }
  h1 {
    padding: ${({ theme }) => theme.spacing(4)} 0;
  }
`;

export const StyledDescription = styled.div`
  flex: 6;
  padding: ${({ theme }) => theme.spacing(4)};
  h2 {
    padding-bottom: ${({ theme }) => theme.spacing(4)};
  }
  ${({ theme }) => theme.breakpoints.sm} {
    flex: 1;
  }
`;
