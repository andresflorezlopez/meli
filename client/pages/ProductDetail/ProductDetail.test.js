import React from "react";
import App from "@config/app";
import ProductDetail from "./ProductDetail";
import renderer from "react-test-renderer";
import LightTheme from "@config/themes/light";
import { ThemeProvider } from "styled-components";

describe("ProductDetail", () => {
  test("should render", () => {
    App.boot();
    const component = renderer.create(
      <ThemeProvider theme={LightTheme}>
        <ProductDetail />
      </ThemeProvider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
