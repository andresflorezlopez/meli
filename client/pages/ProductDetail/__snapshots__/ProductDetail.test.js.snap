// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetail should render 1`] = `
<div>
  <div
    className="sc-18pr4qs-0 eyHxCt"
  >
    <div
      className="sc-1os0qt1-0 ecarEL"
      direction="row"
    >
      <div
        className="sc-18pr4qs-1 jjdFVY"
      >
        <img
          alt="Mercado Libre"
          className="sc-18pr4qs-2 hvNeoX"
          onClick={[Function]}
          src="test-file-stub"
        />
      </div>
      <div
        className="sc-18pr4qs-3 ePKDzW"
      >
        <input
          className="sc-18pr4qs-4 gfnSjD"
          defaultValue=""
          onChange={[Function]}
          onKeyPress={[Function]}
          placeholder="Nunca dejes de buscar"
        />
        <div
          className="sc-18pr4qs-5 bUzMWw"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
  <div
    className="sc-1os0qt1-0 cBGBrT"
    direction="column"
  >
    <div
      className="f30fgq-0 yzvGv"
    >
      <div
        className="sc-1g2dfmx-1 iJYVVG"
      >
        <div
          className="sc-1g2dfmx-0 dFNVXP"
          color="secondary"
        />
      </div>
    </div>
  </div>
</div>
`;
